lib/ts/recipe/thirdpartyemailpassword/recipe.ts

```diff
@@ -0,0 +1,441 @@
+                    },
+                    Recipe.RECIPE_ID
+                );
+            }
+        };
+    }
+
+    static reset() {
+        if (process.env.TEST_MODE !== "testing") {
+            throw new STError(
+                {
+                    type: STError.GENERAL_ERROR,
+                    payload: new Error("calling testing function in non testing env"),
+                },
+                Recipe.RECIPE_ID
+            );
+        }
+        Recipe.instance = undefined;
+    }
+
+    static getInstanceOrThrowError(): Recipe {
+        if (Recipe.instance !== undefined) {
+            return Recipe.instance;
+        }
+        throw new STError(
+            {
+                type: STError.GENERAL_ERROR,
+                payload: new Error("Initialisation not done. Did you forget to call the SuperTokens.init function?"),
+            },
+            Recipe.RECIPE_ID
+        );
+    }
+
+    getAPIsHandled = (): APIHandled[] => {
+        let apisHandled = [
+            ...this.emailPasswordRecipe.getAPIsHandled(),
+            ...this.emailVerificationRecipe.getAPIsHandled(),
+        ];
+        if (this.thirdPartyRecipe !== undefined) {
+            apisHandled.push(...this.thirdPartyRecipe.getAPIsHandled());
+        }
+        return apisHandled;
+    };
+
+    handleAPIRequest = async (id: string, req: express.Request, res: express.Response, next: express.NextFunction) => {
+        if (id === SIGN_UP_API) {
+            return await signUpAPI(this.emailPasswordRecipe, req, res, next);
+        } else if (id === SIGN_IN_API) {
+            return await signInAPI(this.emailPasswordRecipe, req, res, next);
+        } else if (id === GENERATE_PASSWORD_RESET_TOKEN_API) {
+            return await generatePasswordResetTokenAPI(this.emailPasswordRecipe, req, res, next);
+        } else if (id === SIGN_OUT_API) {
+            return await signOutAPI(this.emailPasswordRecipe, req, res, next);
+        } else if (id === PASSWORD_RESET_API) {
+            return await passwordResetAPI(this.emailPasswordRecipe, req, res, next);
+        } else if (id === SIGNUP_EMAIL_EXISTS_API) {
+            return await emailExistsAPI(this.emailPasswordRecipe, req, res, next);
+        }
+        if (this.thirdPartyRecipe !== undefined && id === SIGN_IN_UP_API) {
+            return await signInUpAPI(this.thirdPartyRecipe, req, res, next);
+        } else if (this.thirdPartyRecipe !== undefined && id === AUTHORISATION_API) {
+            return await authorisationUrlAPI(this.thirdPartyRecipe, req, res, next);
+        } else {
+            return await this.emailVerificationRecipe.handleAPIRequest(id, req, res, next);
+        }
+    };
+
+    handleError = (
+        err: STErrorEmailPassword | STErrorThirdParty,
+        request: express.Request,
+        response: express.Response,
+        next: express.NextFunction
+    ): void => {
+        if (err.type === STErrorEmailPassword.EMAIL_ALREADY_EXISTS_ERROR) {
```

Comment:
You can simply call the sub recipes here.. unless there is an error that is shared by both and whose response is different.